pkg/apis/migration/v1alpha1/gvk.go

```diff
@@ -0,0 +1,29 @@
+package v1alpha1
+
+// UnsupportedNamespace - namespace, which is noticed
+// to contain resources unsupported by the migration
+type UnsupportedNamespace struct {
+	Name                 string   `json:"name"`
+	UnsupportedResources []string `json:"unsupportedResources"`
+}
+
+func (p *MigPlanStatus) SelectUnsupported() []string {
+	namespaces := []string{}
+	for _, ns := range p.UnsupportedNamespaces {
+		namespaces = append(namespaces, ns.Name)
+	}
+
+	return namespaces
+}
+
+func (p *MigPlanStatus) InitUnsupported() {
+	p.UnsupportedNamespaces = []UnsupportedNamespace{}
+}
+
+func (p *MigPlanStatus) AppendUnsupported(namespace string, resources []string) {
```

Comment:
Rather than appending the slice in `MigPlanStatus` you can eliminate some complexity and coupling by building a []UnsupportedNamespace{} then assigning to `MigPlanStatus` already populated.